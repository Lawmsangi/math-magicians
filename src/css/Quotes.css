.quotes {
  word-wrap: break-word;
}
