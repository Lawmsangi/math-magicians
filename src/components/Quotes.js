import { useEffect, useState } from 'react';
import '../css/Quotes.css';

export default function Quotes() {
  const [quote, setQuote] = useState([]);
  const [hasError, setHasError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      const options = {
        method: 'GET',
        headers: { 'X-Api-Key': 'OfLmd6NHu3M738oZgj/jBg==h1MlJGB2TSBaKuQo' },
        contentType: 'application/json',
      };
      setIsLoading(true);
      try {
        const res = await fetch('https://api.api-ninjas.com/v1/quotes?catergory=inspirational', options);
        const data = await res.json();
        setQuote(data);
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, [setQuote, setIsLoading]);

  if (hasError) return <div>Something is wrong with your API!</div>;

  if (isLoading) return <div>Loading...wait for a few secondsðŸ˜‰</div>;

  return (
    <div className="quotes">
      {quote.map((item) => (
        <h3 key={item.author}>
          {item.category}
          -
          {' '}
          <span>{item.quote}</span>
        </h3>
      ))}
    </div>
  );
}
