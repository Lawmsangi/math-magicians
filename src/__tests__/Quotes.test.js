import React from 'react';
import { render, waitFor } from '@testing-library/react';
import Quotes from '../components/Quotes';

// Mock the fetch API response
global.fetch = jest.fn(() => Promise.resolve({
  json: () => Promise.resolve([{ category: 'Inspirational', quote: 'Test quote' }]),
}));

test('Quotes component matches snapshot', async () => {
  const { asFragment } = render(<Quotes />);
  await waitFor(() => {
    expect(asFragment()).toMatchSnapshot();
  });
});
