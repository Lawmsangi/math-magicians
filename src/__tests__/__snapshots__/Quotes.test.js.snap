// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Loading...wait for a few secondsðŸ˜‰
  </div>
</DocumentFragment>
`;
